src/common/pages/edit/components/index.js

```diff
@@ -1,6 +1,77 @@
-import React from 'react';
-import PasswordForm from './PasswordForm';
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import PasswordForm from '../../../blocks/PasswordForm';
 
-const Edit = () => <PasswordForm />;
+class EditPage extends PureComponent {
+  state = {
+    name: '',
+    resourceAddress: '',
+    login: '',
+    itemId: null,
+  };
 
-export default Edit;
+  componentDidMount() {
+    const {
+      passwordsItems,
+      match: {
+        params: { id: itemId },
+      },
+    } = this.props;
+
+    const editingPassword = passwordsItems.find(
+      item => item.id.toString() === itemId,
+    );
+    const { name, resourceAddress, login, id } = editingPassword;
+
+    this.setState({
+      itemId: id,
+      name,
+      resourceAddress,
+      login,
+    });
+  }
+
+  submitPasswordForm = values => {
+    const { changePasswordItem } = this.props;
+    const { itemId } = this.state;
+
+    changePasswordItem({ ...values, id: itemId });
+  };
+
+  render() {
+    const { name, resourceAddress, login } = this.state;
+
+    return (
+      <React.Fragment>
+        <PasswordForm
+          name={name}
+          login={login}
+          title="Create Password"
+          resourceAddress={resourceAddress}
+          submitAction={this.submitPasswordForm}
+        />
+      </React.Fragment>
+    );
+  }
+}
+
+EditPage.propTypes = {
+  changePasswordItem: PropTypes.func.isRequired,
+  passwordsItems: PropTypes.arrayOf(
+    PropTypes.shape({
+      createdAt: PropTypes.string,
+      id: PropTypes.number,
+      login: PropTypes.string,
+      name: PropTypes.string,
+      password: PropTypes.string,
+      resourceAddress: PropTypes.string,
+      sendNotificationAt: PropTypes.string,
+      updatedAt: PropTypes.string,
+      userId: PropTypes.number,
+    }),
```

Comment:
Probably make sense to move it into separated file, which will contain Password `PropTypes`. It will be great to split `PropTypes` by feature.
https://labnotes.panderalabs.com/shaping-up-your-react-components-with-proptypes-shape-and-selectors-9d1111d0566f

Reply:
I think it is not required